h2/connection.py

```diff
@@ -586,8 +593,11 @@ def initiate_upgrade_connection(self, settings_header=None):
             For servers, returns nothing.
         :rtype: ``bytes`` or ``None``
         """
-        frame_data = None
+        self.config.logger.debug(
+            "Upgrade connection.\nSettings frame: %s", self.local_settings
```

Comment:
Agreed. Given that we're emitting unstructured logs (@hynek will be so mad at us) anyway, we should probably aim to at least allow users to consider logs as being newline-delimited. Otherwise all the rules of nature are out the window.

As a further note: should we consider [structlog](https://structlog.readthedocs.io/en/stable/)?